src/Processor/Braintree/Hydrator/SubscriptionHydrator.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace TeamGantt\Dues\Processor\Braintree\Hydrator;
+
+use TeamGantt\Dues\Model\Customer;
+use TeamGantt\Dues\Model\PaymentMethod\Token;
+use TeamGantt\Dues\Model\Subscription;
+use TeamGantt\Dues\Processor\Braintree\Repository\CustomerRepository;
+use TeamGantt\Dues\Processor\Braintree\Repository\PaymentMethodRepository;
+use TeamGantt\Dues\Processor\Braintree\Repository\PlanRepository;
+
+class SubscriptionHydrator
+{
+    protected CustomerRepository $customers;
+
+    protected PlanRepository $plans;
+
+    protected PaymentMethodRepository $paymentMethods;
+
+    public function __construct(
+        CustomerRepository $customers,
+        PlanRepository $plans,
+        PaymentMethodRepository $paymentMethods
+    ) {
+        $this->customers = $customers;
+        $this->plans = $plans;
+        $this->paymentMethods = $paymentMethods;
+    }
+
+    /**
+     * Hydrates subscriptions with Plans and Customer info.
+     *
+     * @param Subscription[] $subscriptions
+     */
+    public function hydrate(array $subscriptions): void
+    {
+        $this->hydratePlans($subscriptions);
+        $this->hydrateCustomer($subscriptions);
+    }
+
+    /**
+     * @param Subscription[] $subscriptions
+     */
+    private function hydrateCustomer(array $subscriptions): void
+    {
+        foreach ($subscriptions as $subscription) {
+            $customer = null;
+            $paymentMethod = $subscription->getPaymentMethod();
+
+            if ($paymentMethod instanceof Token) {
+                $customer = $this->getCustomerBySubscriptionPaymentToken($subscription, $paymentMethod);
+            }
+
+            if (null === $customer) {
+                continue;
+            }
+
+            $subscription->setCustomer($customer);
+        }
+    }
+
+    /**
+     * Finds the customer based off of the customer id associated with the payment method and appends it back to the Subscription.
+     */
+    private function getCustomerBySubscriptionPaymentToken(Subscription $subscription, Token $paymentMethod): ?Customer
+    {
+        /**
+         * Customer[].
+         */
+        static $cache = [];
```

Comment:
Updated... I had to tweak things a touch, because it's the calling function that does the loop - which is probably why I did static... but that change does the fix.

You'll see two followup commits that should be fixed back up into their original commits- but that should help with the code review, and I dropped the toString stuff.